calcom/cal.com
# feat: HitPay Payment App

## What does this PR do?

This PR is for HitPay payment app.
We have developed HitPay Payment app that users can make payment for booking using <a href="https://www.hitpayapp.com/">HitPay Payment</a>. 


Loom Video: https://www.loom.com/share/80e85c97c74a4a5b9a6fc7fefe2e78aa?sid=2b191abc-15bc-4378-a129-f95fb8f8541e

## Mandatory Tasks (DO NOT REMOVE)

 
## How should this be tested?

 - Before testing HitPay payments, you need to set up your <a href="https://dashboard.sandbox.hit-pay.com">HitPay sandbox account</a>.
	 <ol>
		<li>If you haven’t already, create an account on <a href="https://dashboard.sandbox.hit-pay.com/register">HitPay</a>.</li>
		<li>Ensure that you have the necessary payment methods <a href="https://docs.hitpayapp.com/introduction#how-to-activate-payment-methods" target="_blank" rel="noreferrer">activated</a> in your HitPay account.</li>
	 </ol>  

- Are there environment variables that should be set?
  These 2 environment variables should be set.
  NEXT_PUBLIC_API_HITPAY_PRODUCTION='https://api.hit-pay.com'
  NEXT_PUBLIC_API_HITPAY_SANDBOX='https://api.sandbox.hit-pay.com'

1.  There is HitPay payment app on /apps and /apps/categories/payment.
2.  Install HitPay payment app on /apps/hitpay   
 
      ![2024-10-19_10h34_47](https://github.com/user-attachments/assets/5c34f0b5-6326-421b-8d9b-8dffe014c46e)

3.  Enter API Key and Salt values under Account Information and click on Connect with HitPay on /apps/hitpay/setup page. 
You can get API keys from HitPay Dashboard >API Keys.
If you check sandbox switch next to Account Information, you can enter the API key and Salt of HitPay sanbox and all the transactions on events with HitPay enabled will be handled over HitPay sandbox.

     ![2024-10-19_10h41_47](https://github.com/user-attachments/assets/f3287d26-79ed-4097-9cca-25ded757affc)

4.  Check if the API key and Salt are updatable on /apps/hitpay/setup page after changing any value. The values cannot be edited at hidden status.
5.  Install HitPay app on an event.

     ![2024-10-21_10h44_50](https://github.com/user-attachments/assets/471d0051-ae48-44b5-b616-2711e53f083f)

6.  Check if the HitPay payment is working correctly on a booking page and then if the transaction is on Transactions page on the HitPay dashboard.

    ![2024-10-21_10h54_24](https://github.com/user-attachments/assets/a51d9488-98ca-4bad-b843-6fcc8eaa9566)
